team/mantras.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Team mantras
+description: A set of high level mantras the web and design team embrace
+---
+
+## Never remove data or functionality because of screen size
+Never hide information or remove functionally on different screen sizes. In the modern era,  applications should be fully featured and easy to use on all devices and screens.
+
+**Do:** If a details page contains a set of metadata and actions which can be performed on that entity. That data and functionality should be accessible on all screens.
+For example, when viewing the list on a phone the user should be able to search and filter as they can on a desktop screen.
+
+**Don’t**: Reduce the information on the listing to fit the content in.
+For example, tables rows may need to stack or become expandible to persist that same information.
+
+
+## Do not assume a user's intent
+Any request for data should be clearly triggered by a user action such as a click or key press. Scrolling or hovering is not an intentional data request from a user.
+
+**Do**: Provide a button or interaction that will clearly request more data from the server.
+For example, provide a “Load more…” button at the end of a limited list to request to next page of data.
+
+**Don’t**: On mouseover or scroll request additional data from the server.
+For example: a tooltip that requests extra data to populate its contents.
+
+
+## Reflect state of the application in the URL
+All views in an application should be reflected in the URL so a user can share or bookmark an area and jump to it on load. This includes filter and search inputs on listing pages.
+
+**Do**:  When entering an entity details reflect that view in the URL.
+For example, /entity/12345
+
+**Don’t**: When clicking to reveal a popover update the address. For example, when filtering and selecting the filters from the drop down do not persist that state in the URL.
+For example, /listing/+filter-open
+
+
+## Always allow the user to undo an action
```

Comment:
Are these mantras retroactive? If so, this would require significant changes to both MAAS and RBAC (and possibly other apps of ours).

Reply:
Sure, as guidelines for all patterns and interactions going forward.

Reply:
Yeah, I think it's fine for it to be a principle without us having successfully followed it everywhere. Going forward, we should make every effort to follow it, but there may still be (hopefully extremely rare) exceptions.

Reply:
Undo is awesome (much better than a confirmation step), but sometimes it’s less awesome than other stuff we could implement in the same amount of time. So I think “Always” is too strong here.

Separately, many things can’t or shouldn’t be undoable. For example, formatting or repartitioning a disk, or deleting your PII from a service. One common substitute is providing a timeout. For example, you can’t undo sending e-mail, but in Gmail you get a few seconds to “Undo” (cancel) it instead. You can’t undo deleting your Facebook account, but you get 30 days to change your mind. And often you can’t undo deleting a file/message/photo, but instead items in a trash may expire after a week/month.

Reply:
I still think this is a good principle, and it can lead us to consider implementing interesting solutions like the Gmail delays for non-retractable / descructive actions like formatting.

Agree "always" is probably the wrong word.